### PR TITLE
add api doc

### PR DESCRIPTION
**Pull request by morri-son**
**What this PR does / why we need it**:
- Add minimalistic API document and link it (we may think of removing the complete link / basic api.md as the content is part of the readme)

- Made minor format changes to the README.

### CODE REVIEW

File: api.md [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+
+- `400 Bad Request`: Ungültige Anfrage
+- `401 Unauthorized`: Nicht autorisiert
+- `500 Internal Server Error`: Serverfehler
+
+### Beispiele
+
+**Beispiel für eine Signaturanfrage**
+
+*Anfrage:**
```
Comment (8R0WNI3) [inline api.md]: nit: Missing `*` at the beginning.

File: api.md [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+```json
+{
+  "data": "Hello, World!",
+  "hashAlgorithm": "SHA256",
+  "encoding": "utf-8"
+}
+```
+
+<vscode_annotation details='%5B%7B%22title%22%3A%22hardcoded-credentials%22%2C%22description%22%3A%22Embedding%20credentials%20in%20source%20code%20risks%20unauthorized%20access%22%7D%5D'></vscode_annotation>
```
Comment (8R0WNI3) [inline api.md]: What is the purpose of this `vscode_annotation`?

**Review by 8R0WNI3 — COMMENTED** [review]
General question, shouldn't the api documentation rather be written in English?

File: api.md (hunk 1 above)
Comment (morri-son) [reply api.md]: corrected

File: api.md (hunk 2 above)
Comment (morri-son) [reply api.md]: removed, leftover from Mr. Copilot

**Comment by morri-son** [comment]
> General question, shouldn't the api documentation rather be written in English?

Totally correct. I've created the doc using Github Copilot. Seems that I asked to create an api.md in German. And even more strange that I went over the doc before upload and did not even recognised it 😂. I corrected this now and uploaded an English version.

**Review by 8R0WNI3 — DISMISSED** [review]
/lgtm

**Review by 8R0WNI3 — APPROVED** [review]

**Pull request merged**